profiling/src/allocation.rs

```diff
@@ -180,11 +183,9 @@ pub fn allocation_profiling_rinit() {
 
     // `is_zend_mm()` should be `false` now, as we installed our custom handlers
     if is_zend_mm() {
-        error!("Memory allocation profiling could not be enabled. Please feel free to fill an issue stating the PHP version and installed modules. Most likely the reason is your PHP binary was compiled with `ZEND_MM_CUSTOM` being disabled.");
-        REQUEST_LOCALS.with(|cell| {
-            let mut locals = cell.borrow_mut();
-            locals.profiling_allocation_enabled = false;
-        });
+        // Can't proceed with it being disabled, because that's a system-wide
+        // setting, not per-request.
+        panic!("Memory allocation profiling could not be enabled. Please feel free to fill an issue stating the PHP version and installed modules. Most likely the reason is your PHP binary was compiled with `ZEND_MM_CUSTOM` being disabled.");
```

Comment:
I have to say that I like this way more. It is very intrusive, but OTOH folks would not ask questions why they don't see allocation profiles in the backend.